### ISSUE TITLE
[Q] How to support other DL backend like paddle?

### ISSUE THREAD

**Issue by HydrogenSulfate** [wandb/wandb#5854]
I found wandb is efficient and easy to use, and I want to add [paddlepaddle](https://www.paddlepaddle.org.cn/en) to wandb like pytorch/tensorflow does, so how can I contribute that?

**Comment by thanos-wandb** [comment]
Hi @HydrogenSulfate thank you for contacting us and glad to hear you would be interested to contribute integrating `wandb` with other frameworks. We currently have an integration with [PaddleOCR](https://docs.wandb.ai/guides/integrations/paddleocr) and [PaddleDetection](https://docs.wandb.ai/guides/integrations/paddledetection) implemented in `PaddlePaddle`. Have you checked the `WandbCallbacks` in Paddle's repo, would this not work for your use case?

We also have a [guide in our Docs page ](https://docs.wandb.ai/guides/integrations/add-wandb-to-any-library) on how to integrate `wandb` with other libraries. For contributing to our SDK, please also refer to [our guide here](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md). I hope this helps, but please let us know if you had any more questions.

**Comment by thanos-wandb** [comment]
Sure! Thanks @HydrogenSulfate for the update, and please feel free to reach out to us back here if you didn't find the information you were looking for.

**Comment by HydrogenSulfate** [comment]
> 

I have implemented a easy use of wandb in PaddleScience in that PaddlePaddle/PaddleScience#422, and I am going to add it to docs in wandb like PaddleOCR...

**Comment by HydrogenSulfate** [comment]
> Sure! Thanks @HydrogenSulfate for the update, and please feel free to reach out to us back here if you didn't find the information you were looking for.

I have implemented a easy usage of wandb in PaddleScience in https://github.com/PaddlePaddle/PaddleScience/pull/422 with reference to wandb's document, and I am going to add it to docs in wandb like PaddleOCR...

**Comment by thanos-wandb** [comment]
Great to hear that, thanks so much for the contribution! Please keep us posted here once it's merged.

**Comment by HydrogenSulfate** [comment]
> Great to hear that, thanks so much for the contribution! Please keep us posted here once it's merged.

I have make wandb in PaddleScience：https://paddlescience-docs.readthedocs.io/zh/latest/zh/user_guide/#14-wandb

But i find a problem as descripted in the user guide, wandb seems use `Step` for default counter which will be added 1 when we call `wandb.log`, but not every one's code would prepare all stats for logging and call `wandb.log` onece per step, such as me. I wound call several `wandb.log` for not all my stats is prepared at the same time. So I note in the guide latter for change the x-axis from `Step` to `step` for correct the step number. Is there any good solution for this?

**Comment by ayulockin** [comment]
Hey @HydrogenSulfate you can use the `commit=False` inside your `wandb.log` calls.

Example:

```
for _ in range(10):
    # do something
    wandb.log({"fancy_metric": metric1}, commit=False)
    # do something more
    wandb.log({"fancy_metric1": metric2}, commit=False)
    # final do something
    wandb.log({"fancy_metric2": metric3}, commit=True)
```

`commit=True` is default so you can choose to not pass it explicitly.

**Comment by HydrogenSulfate** [comment]
> 

Thanks for replying, I guess it might be not convenient to know when is comes to a final log call. Your code might be somewhat a simple case but a generic case for those git repo which encapsulate common functions into utils. It is hard to know when a log call is a final call except for adding an extra flag argument like `commit: bool = False` in my fnction.

``` py
def scaler(name: str, value: float, step: int, vdl_writer=None, wandb_writer=None):
    if vdl_writer is not None:
        vdl_writer.add_scalar(tag=name, step=step, value=value)
    if wandb_writer is not None:
        wandb_writer.log({"step": step, name: value})
```

It may still be the best solution with changing axis to `step`😓
